src/main/git/gitAuthorization.ts

```diff
@@ -0,0 +1,32 @@
+import axios, {AxiosResponse} from 'axios'
+import {EndPointParameters, Url} from "./gitTypes";
+
+export function authorizeWithGithub() : void {
+    axios.get(buildAuthorizeEndpoint()).then(data => {
+        document.location.href = data.config.url;
+    });
+}
+
+/**
+ *
+ * @param codeParam
+ */
+export async function postAccessToken(codeParam : string) : Promise<AxiosResponse> {
+    return await axios.post(buildAccessTokenEndpoint(codeParam)).then(data => {
+        const search = data.data.substring(0);
+        const result = JSON.parse('{"' + search.replace(/&/g, '","').replace(/=/g,'":"') + '"}', function(key, value) { return key===""?value:decodeURIComponent(value) })
+        return result.access_token;
+    });
+}
+
+function buildAuthorizeEndpoint() : string {
+    let clientIdParam = "?"+ EndPointParameters.CLIENT_ID + "=" + process.env.GITHUB_CLIENT_ID;
```

Comment:
just a suggestion that JS has some built in functionalities to handle URL params. no need to change right now, but worth knowing:  
https://developer.mozilla.org/en-US/docs/Web/API/URL
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams